content/docs/concepts/network/index.md

```diff
@@ -0,0 +1,437 @@
+use virtual services to finely configure traffic. For example, to direct HTTP
+traffic to use a different version of the service for a specific user. To
+specify the user, the following example uses a regular expression to match the
+user from the cookie and shows you how to add the condition for the protocol in
+the `http:` field and a new `route:` and `destination:` configuration for the
+traffic matching the condition. In short, the example configures the HTTP
+traffic from the `jason` user to go to the `v2` subset of the `my-svc` service:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: my-vtl-svc
+spec:
+  hosts:
+    - *
+    - route:
+      - destination:
+        host: my-svc
+        subset: v1
+  http:
+    - match:
+      - headers:
+          cookie:
+            regex: "^(.*?;)?(user=jason)(;.*)?$"
+    route:
+      - destination:
+        host: my-svc
+        subset: v2
+{{< /text >}}
+
+The following diagram shows the configured traffic rules for both the traffic
+with the matched cookie and for all other traffic:
+
+{{< image width="50%"
+    link="./net-config-4.svg"
+    caption="Configurable traffic rules for traffic with and without a matched cookie"
+    >}}
+
+If you are working with a Kubernetes-based mesh, use virtual services to manage
+the traffic to your namespaces. The following example shows the `my-namespace`
+virtual service. It configures traffic routes for the two services behind the
+`my-namespace.svc.cluster.local` Kubernetes namespace:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: my-namespace
+spec:
+  hosts:
+    - *.my-namespace.svc.cluster.local
+  http:
+  - match:
+    - uri:
+        prefix: /svc-1
+    route:
+    - destination:
+        host: svc-1
+  - match:
+    - uri:
+        prefix: /svc-2
+    route:
+    - destination:
+        host: svc-2
+{{< /text >}}
+
+The following diagram shows the configured traffic routes implementing a
+virtual service to configure traffic for services in a namespace for two
+distinct application services based on the URI prefixes:
+
+{{< image width="50%"
+    link="./net-config-5.svg"
+    caption="Configurable traffic rules implementing a namespace for two  application services"
+    >}}
+
+Once you have identified and configured the virtual services suitable for your
+mesh, you can add gateways to route traffic in or out of your mesh. The virtual
+services together with destination rules allow you to configure the behavior of
+the traffic to fulfill the needs of your mesh. Together with a service entry,
+you can use virtual services to configure traffic routes to external
+dependencies.
+
+Visit our [virtual services reference documentation](/docs/reference/config/istio.networking.v1alpha3/#VirtualService) to review all the enabled keys and values.
+
+## Gateways
+
+The Istio gateways are layer 7 load balancers. With an Istio gateway, you can
```

Comment:
Is it correct to say that every configuration object with `kind: Gateway` requires 1 or more workloads to work? How can I identify the workload in the configuration object? You mention that they only configure layer 4-6 but I read [here](https://istio.io/blog/2018/v1alpha3-routing/#gateway) that they implement L7. Can you help me by clarifying further?

Reply:
Right, so this is the problem. A "gateway" is a workload, an L7 router implementation (typically an Envoy running in a pod). A `Gateway`, which is what you are talking about here, is the kind of the configuration resource that configures the L4-6 properties of the gateway that it is bound to. It's bound to a particular gateway (workload) with the `selector` field in the `Gateway`. A `VirtualService` is then used to configure the remaining L7 properties of the gateway. This is why I think its best to explicitly refer to the configuration resources by the value of there kind field. That seems pretty clear to me.

Reply:
Btw, the style guide doesn't really address the issue of the configuration resource kinds although it does have examples of referring to them as I suggest: https://istio.io/about/contribute/style-guide/#use-code-style-for-inline-code-and-commands

Notice the word `Deployment`.

Reply:
@frankbu Thank you for the clarification. I've modified the text using your outstanding explanation. To avoid the terminology issue, I'm using the term "gateway configuration object" to distinguish the actual "gateway workload" 
IMO using the value of `kind:` doesn't really solve the problem of us referring to two very different things with the same term, it just transforms it into a markup use consistency issue. 
The best practice in technical writing is clear here: https://docs.microsoft.com/en-us/style-guide/word-choice/use-technical-terms-carefully
Having some specific research on this might be helpful but until then, I'll use the workaround described above.